### PR TITLE
OCPBUGS-25470: Fix CI test failures

### PR DESCRIPTION
**Pull request by nishant-parekh**
Match ginkgov2 version used for tests to the vendored version
Update golang version used to build tests

### CODE REVIEW

File: hack/run-functests.sh [hunk 1]
```diff
@@ -22,9 +28,10 @@ export PATH=$PATH:$GOPATH/bin
...
 
+go version
 if [[ $SUITE == *"$VALIDATION_SUIT_SUBSTR"* ]]
 then
 	GOFLAGS=-mod=vendor ginkgo --output-dir=$JUNIT_OUTPUT_DIR --junit-report=$JUNIT_OUTPUT_FILE -v -p "$SUITE"
 else
-	GOFLAGS=-mod=vendor ginkgo -keepGoing --output-dir=$JUNIT_OUTPUT_DIR --junit-report=$JUNIT_OUTPUT_FILE -v -p "$SUITE"/serial "$SUITE"/parallel
-fi
\ No newline at end of file
+	GOFLAGS=-mod=vendor ginkgo -keep-going --output-dir=$JUNIT_OUTPUT_DIR --junit-report=$JUNIT_OUTPUT_FILE -v -p "$SUITE"/serial "$SUITE"/parallel
```
Comment (edcdavid) [inline hack/run-functests.sh:36]: -keep-going -> --keep-going

File: hack/run-functests.sh [hunk 2]
```diff
@@ -1,15 +1,21 @@
 #!/bin/bash
 set -x
-which ginkgo
+# Install Go 1.20.4
+wget https://go.dev/dl/go1.20.4.linux-amd64.tar.gz
+rm -rf /usr/local/go
+tar -C /usr/local -xzf go1.20.4.linux-amd64.tar.gz
+export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin
```
Comment (edcdavid) [inline hack/run-functests.sh:7]: Why 2 PATH lines? why not just: 
`export PATH=$PATH:/usr/local/go/bin`

**Review by edcdavid — APPROVED** [review]
Looks good to me, just a few comments

File: hack/run-functests.sh [hunk 3]
```diff
@@ -1,15 +1,21 @@
 #!/bin/bash
 set -x
-which ginkgo
+# Install Go 1.20.4
+wget https://go.dev/dl/go1.20.4.linux-amd64.tar.gz
+rm -rf /usr/local/go
```
Comment (josephdrichard) [inline hack/run-functests.sh:5]: So this will remove existing go binary on a user's system?

Comment (nishant-parekh) [reply hack/run-functests.sh:5]: The version of ginkgo used below requires at least this version of golang. This change will enable the CI setup to run the tests as the image used for CI has an older version of golang.. there is an ongoing effort to change the image used after which this change can be rolled back

File: hack/run-functests.sh [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
```
Comment (jzding) [inline hack/run-functests.sh:5]: If this is a temporary workaround please add a `TODO` comment. For a permanent solution, I suggest 1. check the existing go version and install newer version only when needed. 2. install new go in {ptp-operator repo}/bin and add this in front of the PATH.

**Review by jzding — APPROVED** [review]
/lgtm

File: hack/run-functests.sh (hunk 4 above)
Comment (nishant-parekh) [reply hack/run-functests.sh:5]: I will have a subsequent update to address your comment. Trying to enable CI pipeline again with this change.

**Pull request merged**